oddt/shape.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+import scipy.stats as stats
+
+
+def standard_usr(molecule, points=[None, None, None, None]):
+    """Computes USR shape descriptor
+
+    Parameters
+    ----------
+    molecule : oddt.toolkits.ob.Molecule
+
+    points : array
+        Coordinates
+
+    Returns
+    -------
+        shape_descriptor : numpy array
+    """
+    atoms = molecule.atom_dict['coords']
+    ctd, fct, cst, ftf = points
+
+    "ctd: the molecular centroid"
+    if not ctd: ctd = atoms.mean(0)
+    distances_ctd = np.linalg.norm(atoms - ctd, axis=1)
+
+    "cst: the closest atom to the molecular centrid"
+    if not cst: cst = atoms[distances_ctd.argmin()]
+    distances_cst = np.linalg.norm(atoms - cst, axis=1)
+
+    "fct: the farthest atom to the molecular centroid"
+    if not fct: fct = atoms[distances_ctd.argmax()]
+    distances_fct = np.linalg.norm(atoms - fct, axis=1)
+
+    "ftf: the farthest atom to the farthest atom to the molecular centroid"
+    if not ftf: ftf = atoms[distances_fct.argmax()]
+    distances_ftf = np.linalg.norm(atoms - ftf, axis=1)
+
+    distances_list = [distances_ctd, distances_cst, distances_fct, distances_ftf]
+
+    shape_descriptor = np.empty(12)
+
+    i = 0
+    for distances in distances_list:
+        shape_descriptor[0 + i] = np.mean(distances)
+        shape_descriptor[1 + i] = np.var(distances)
+        shape_descriptor[2 + i] = stats.skew(distances)
+        i += 3
+
+    return shape_descriptor, [cst, fct, cst, ftf]
+
+
+def usr_cat(molecule):
+    """Computes USRCAT shape descriptor
+
+    Parameters
+    ----------
+    molecule : oddt.toolkits.ob.Molecule
+    """
+    all_atoms = molecule.atom_dict['coords']
+    hydrophobic = molecule.atom_dict[molecule.atom_dict['ishydrophobe']]
+    aromatic = molecule.atom_dict[molecule.atom_dict['isaromatic']]
+    acceptor = molecule.atom_dict[molecule.atom_dict['isacceptor']]
+    donor = molecule.atom_dict[molecule.atom_dict['isdonor']]
+
+    p1, points = standard_usr(all_atoms)
```

Comment:
repeated code, you can use a list or a dict here (with 'ishydrophobe' etc as keys) instead of p2-p5